mantle/cmd/kola/iscsi.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"github.com/coreos/coreos-assembler/mantle/platform/conf"
+)
+
+// from https://github.com/jbtrystram/targetcli-containers/tree/main/ignition
+var iscsiTargetConfig = conf.Butane(`
+variant: fcos
+version: 1.5.0
+storage:
+  filesystems:
+    - path: /var
+      device: /dev/disk/by-id/virtio-var
+      format: ext4
+      wipe_filesystem: true
+      label: var
+      with_mount_unit: true
+  files:
+    - path: /etc/containers/systemd/target.container
+      contents:
+        inline: |
+            [Unit]
+            Description=Targetd container
+            Documentation=https://github.com/jbtrystram/targetcli-containers
+            After=local-fs.target
+            After=network-online.target, After=nss-lookup.target, dev-disk-by\x2did-virtio\x2dtarget.device
+            Wants=network-online.target
+            [Container]
+            Image=quay.io/jbtrystram/targetcli:latest
+            ContainerName=target
+            AddCapability=CAP_SYS_MODULE
+            Network=host
+            Volume=/dev/disk/by-id/virtio-target:/dev/disk/by-id/virtio-target
```

Comment:
I tried to use `AddDevice` but unfortunately podman `--device` resolves the symlink, so we get `/dev/vd?` mounted into the container.

The `/dev/disk/by-*` shortcuts are not recreated into the container, so we are left to make a guess. I'd rather have the link passed as a volume to be explicit rather than implicit.

Reply:
I tested yesterday and could not find them in the container. maybe we need to pass other options to podman